crates/red_knot_python_semantic/src/semantic_index/builder.rs

```diff
@@ -98,6 +104,10 @@ pub(super) struct SemanticIndexBuilder<'db> {
     expressions_by_node: FxHashMap<ExpressionNodeKey, Expression<'db>>,
     imported_modules: FxHashSet<ModuleName>,
     eager_bindings: FxHashMap<EagerBindingsKey, ScopedEagerBindingsId>,
+    python_version: PythonVersion,
+    semantic_checker: SemanticSyntaxChecker,
```

Comment:
We should move those fields into their own group (or to the `builder state` group at the top). The fields at the bottom are reserved for fields that match the fields on `SemanticIndex`